src/server/validateDependencies.ts

```diff
@@ -110,27 +118,40 @@ async function executablesOrSharedLibraries(directoryPath: string): Promise<stri
 }
 
 async function missingFileDependencies(filePath: string): Promise<Array<string>> {
-  const {stdout} = await lddAsync(filePath);
-  const missingDeps = stdout.split('\n').map(line => line.trim()).filter(line => line.endsWith('not found') && line.includes('=>')).map(line => line.split('=>')[0].trim());
-  return missingDeps;
-}
-
-function lddAsync(filePath: string): Promise<{stdout: string, stderr: string, code: number}> {
   const dirname = path.dirname(filePath);
-  const ldd = spawn('ldd', [filePath], {
+  const {stdout, code} = await spawnAsync('ldd', [filePath], {
     cwd: dirname,
     env: {
       ...process.env,
       LD_LIBRARY_PATH: process.env.LD_LIBRARY_PATH ? `${process.env.LD_LIBRARY_PATH}:${dirname}` : dirname,
     },
   });
+  if (code !== 0)
```

Comment:
We should throw if it failed otherwise it's indistinguishable from the case when all dependencies are good.

Reply:
`ldd` fails if we feed in a non-ELF file (and we might occasionally do so). 

> We should throw if it failed otherwise it's indistinguishable from the case when all dependencies are good.

If our spawn fails for some reason, i'd still try launching to avoid false negatives.

Reply:
That's unfortunate, perhaps we could run this checks after launch failed? Don't remember what was the rationale for running the checks before launch.

Reply:
The reason is that there might be no crash at all. In case of missing `libglesv2` the browsers won't crash - instead, renderer will either misbehave (on ubuntu 18.04) or crash (on ubuntu 20.04)